### DISCUSSION TITLE
try_convert() or begin try equivalent

### DISCUSSION THREAD

**Discussion by nottinghamgoclub** [duckdb/duckdb#3804] [General]
Taking a use case, in staging, I have a column that should be dates, but the data is dirty. Not all the dates are in the correct format, and a few are text values (e.g. "date missing"). I want this column to be a date column in the business layer so that I can use date functions, work out differences, group by months, and so on. 

Going through this scenario, when inserting into a table in staging, I would create the table with this troublesome column as a varchar, because otherwise it would throw an error when importing the CSV file. When importing the csv with the auto format flag, it would also format the column as a varchar.

Then comes the job of managing the data between the dirty layer and the clean layer. If I were using SQL Server, I would then use something like try_convert(), or use a begin try, to convert the dates that I can convert, and ignore the dates that cannot be converted. Or, test for an exception / a default value, using an if statement, and deal with the malformatted dates separately. It may be that only 2 records out of 50,000 are not able to be parsed, but unless I use some error handling, the process would fall over. 

What is the equivalent handling in duckdb especially when converting data formats so that procedures can continue to execute?

**Comment by Mytherin** [comment]
You’re probably looking for TRY_CAST(x AS DATE)

**Comment by Alex-Monahan** [comment]
I have run into this in the past with `strptime`. It fails on unclean data rather than returning NULL.
